src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADTaskManager.java

```diff
@@ -832,7 +950,9 @@ private void onIndexADTaskResponse(
             // Put detector id in cache. If detector id already in cache, will throw
             // DuplicateTaskException. This is to solve race condition when user send
             // multiple start request for one historical detector.
-            adTaskCacheManager.add(adTask.getDetectorId());
+            if (adTask.isHistoricalTask()) {
```

Comment:
Curiously, the realtime task doesn't need to be added to the TaskCacheManager?
If it's true, how about to rename the TaskCacheManager to HistoricalTaskCacheManager in future?

Reply:
Will add some realtime task cache in later PRs. Realtime and historical task are using different code path now. It's another topic to merge these code path. Will be easier to develop and maintain after unifying the code.

Reply:
Just update: add realtime task cache in this PR #130, check `ADTaskCacheManager.java`